diff --git a/drivers/staging/qca-wifi-host-cmn/qdf/linux/src/qdf_lock.c b/drivers/staging/qca-wifi-host-cmn/qdf/linux/src/qdf_lock.c
index 48deb01..48adc60 100644
--- a/drivers/staging/qca-wifi-host-cmn/qdf/linux/src/qdf_lock.c
+++ b/drivers/staging/qca-wifi-host-cmn/qdf/linux/src/qdf_lock.c
@@ -241,19 +241,10 @@ qdf_export_symbol(qdf_mutex_release);
  *
  * Return: Pointer to the name if it is valid or a default string
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0))
-const char *qdf_wake_lock_name(qdf_wake_lock_t *lock)
-{
-	if (lock)
-		return lock->lock.name;
-	return "UNNAMED_WAKELOCK";
-}
-#else
 const char *qdf_wake_lock_name(qdf_wake_lock_t *lock)
 {
 	return "NO_WAKELOCK_SUPPORT";
 }
-#endif
 qdf_export_symbol(qdf_wake_lock_name);
 
 /**
@@ -265,34 +256,10 @@ qdf_export_symbol(qdf_wake_lock_name);
  * QDF status success: if wake lock is initialized
  * QDF status failure: if wake lock was not initialized
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 110)) || \
-	defined(WAKEUP_SOURCE_DEV)
-QDF_STATUS qdf_wake_lock_create(qdf_wake_lock_t *lock, const char *name)
-{
-	qdf_mem_zero(lock, sizeof(*lock));
-	lock->priv = wakeup_source_register(lock->lock.dev, name);
-	if (!(lock->priv)) {
-		QDF_BUG(0);
-		return QDF_STATUS_E_FAILURE;
-	}
-
-	lock->lock = *(lock->priv);
-
-	return QDF_STATUS_SUCCESS;
-}
-#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0))
 QDF_STATUS qdf_wake_lock_create(qdf_wake_lock_t *lock, const char *name)
 {
-	wakeup_source_init(&(lock->lock), name);
-	lock->priv = &(lock->lock);
 	return QDF_STATUS_SUCCESS;
 }
-#else
-QDF_STATUS qdf_wake_lock_create(qdf_wake_lock_t *lock, const char *name)
-{
-	return QDF_STATUS_SUCCESS;
-}
-#endif
 qdf_export_symbol(qdf_wake_lock_create);
 
 /**
@@ -304,22 +271,10 @@ qdf_export_symbol(qdf_wake_lock_create);
  * QDF status success: if wake lock is acquired
  * QDF status failure: if wake lock was not acquired
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0))
 QDF_STATUS qdf_wake_lock_acquire(qdf_wake_lock_t *lock, uint32_t reason)
 {
-	host_diag_log_wlock(reason, qdf_wake_lock_name(lock),
-			    WIFI_POWER_EVENT_DEFAULT_WAKELOCK_TIMEOUT,
-			    WIFI_POWER_EVENT_WAKELOCK_TAKEN);
-	__pm_stay_awake(lock->priv);
-
 	return QDF_STATUS_SUCCESS;
 }
-#else
-QDF_STATUS qdf_wake_lock_acquire(qdf_wake_lock_t *lock, uint32_t reason)
-{
-	return QDF_STATUS_SUCCESS;
-}
-#endif
 qdf_export_symbol(qdf_wake_lock_acquire);
 
 /**
@@ -331,27 +286,10 @@ qdf_export_symbol(qdf_wake_lock_acquire);
  * QDF status success: if wake lock is acquired
  * QDF status failure: if wake lock was not acquired
  */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
 QDF_STATUS qdf_wake_lock_timeout_acquire(qdf_wake_lock_t *lock, uint32_t msec)
 {
-	pm_wakeup_ws_event(lock->priv, msec, true);
 	return QDF_STATUS_SUCCESS;
 }
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
-QDF_STATUS qdf_wake_lock_timeout_acquire(qdf_wake_lock_t *lock, uint32_t msec)
-{
-	/* Wakelock for Rx is frequent.
-	 * It is reported only during active debug
-	 */
-	__pm_wakeup_event(&(lock->lock), msec);
-	return QDF_STATUS_SUCCESS;
-}
-#else /* LINUX_VERSION_CODE */
-QDF_STATUS qdf_wake_lock_timeout_acquire(qdf_wake_lock_t *lock, uint32_t msec)
-{
-	return QDF_STATUS_SUCCESS;
-}
-#endif /* LINUX_VERSION_CODE */
 qdf_export_symbol(qdf_wake_lock_timeout_acquire);
 
 /**
@@ -363,22 +301,10 @@ qdf_export_symbol(qdf_wake_lock_timeout_acquire);
  * QDF status success: if wake lock is acquired
  * QDF status failure: if wake lock was not acquired
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0))
-QDF_STATUS qdf_wake_lock_release(qdf_wake_lock_t *lock, uint32_t reason)
-{
-	host_diag_log_wlock(reason, qdf_wake_lock_name(lock),
-			    WIFI_POWER_EVENT_DEFAULT_WAKELOCK_TIMEOUT,
-			    WIFI_POWER_EVENT_WAKELOCK_RELEASED);
-	__pm_relax(lock->priv);
-
-	return QDF_STATUS_SUCCESS;
-}
-#else
 QDF_STATUS qdf_wake_lock_release(qdf_wake_lock_t *lock, uint32_t reason)
 {
 	return QDF_STATUS_SUCCESS;
 }
-#endif
 qdf_export_symbol(qdf_wake_lock_release);
 
 /**
@@ -389,25 +315,10 @@ qdf_export_symbol(qdf_wake_lock_release);
  * QDF status success: if wake lock is acquired
  * QDF status failure: if wake lock was not acquired
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 110)) || \
-	defined(WAKEUP_SOURCE_DEV)
-QDF_STATUS qdf_wake_lock_destroy(qdf_wake_lock_t *lock)
-{
-	wakeup_source_unregister(lock->priv);
-	return QDF_STATUS_SUCCESS;
-}
-#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0))
 QDF_STATUS qdf_wake_lock_destroy(qdf_wake_lock_t *lock)
 {
-	wakeup_source_trash(&(lock->lock));
 	return QDF_STATUS_SUCCESS;
 }
-#else
-QDF_STATUS qdf_wake_lock_destroy(qdf_wake_lock_t *lock)
-{
-	return QDF_STATUS_SUCCESS;
-}
-#endif
 qdf_export_symbol(qdf_wake_lock_destroy);
 
 /**
