diff --git a/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_wext.c b/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_wext.c
--- a/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_wext.c
+++ b/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_wext.c
@@ -11067,3 +11067,5 @@
 static const iw_handler we_standard[] = {
 	[IW_IOCTL_IDX(SIOCGIWFREQ)] = (iw_handler)hdd_wext_giwfreq,
+	[IW_IOCTL_IDX(SIOCSIWMODE)] = (iw_handler)cfg80211_wext_siwmode,
+	[IW_IOCTL_IDX(SIOCGIWMODE)] = (iw_handler)cfg80211_wext_giwmode,
 };
