diff --git a/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_main.c b/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_main.c
--- a/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_main.c
+++ b/drivers/staging/qcacld-3.0/core/hdd/src/wlan_hdd_main.c
@@ -2769,5 +2769,13 @@
 	if (!ret)
 		ret = hdd_enable_monitor_mode(dev);
 
+#ifdef FEATURE_FRAME_INJECTION_SUPPORT
+	/* Init injection_ctx for mission-mode VIF */
+	if (!ret && !adapter->injection_ctx) {
+		if (QDF_IS_STATUS_ERROR(hdd_init_frame_injection(adapter)))
+			hdd_err("Failed to init injection for mon VIF");
+	}
+#endif
+
 	if (ret)
 		return ret;
